frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -292,9 +292,15 @@ void retry() {
             return;
         }
 
-        logger.info("Retry worker: {}", workerId);
-        ModelManager modelManager = ModelManager.getInstance();
-        modelManager.submitTask(this);
+        ModelManager manager = ModelManager.getInstance();
+
+        this.backoffIdx = (backoffIdx + 1) % Model.BACKOFF.length;
```

Comment:
Sorry about earlier comment. This isn't correct:
if (backoffIdx < Model.BACKOFF.length -1) 
    ++backoffIdx;
}